server/src/main/java/org/elasticsearch/index/IndexNotFoundException.java

```diff
@@ -46,7 +46,7 @@ public IndexNotFoundException(Index index) {
     }
 
     public IndexNotFoundException(Index index, Throwable cause) {
-        super("no such index", cause);
+        super("no such index " + index.toString(), cause);
```

Comment:
nit: `"no such index [" + index.getName() + "]"` as `index.toString()` could return smth like `[index42/n6gzFZTgS664GUfx0Xrpjw]` that is a bit inconsistent with other messages